scrapy/utils/conf.py

```diff
@@ -21,7 +20,7 @@
 
 def build_component_list(
     compdict: MutableMapping[Any, Any],
-    custom: Any = None,
+    *,
```

Comment:
While this seems similar to the `convert_image` scenario above, I feel like here we are more justified in this change, because the Pythonic way to fix `build_component_list({}, foo, bar)` was switching to `build_component_list({}, convert=bar)`.

That said, I wonder if we should follow a similar deprecation approach, in case some users went with `build_component_list({}, None, bar)` instead, as that was also a valid way to get rid of the warning.